spin/cmds/meson.py

```diff
@@ -339,29 +342,72 @@ def build(
 
         # Any other conditions that warrant a reconfigure?
 
+    meson_args_compile = list(meson_args.get("compile", tuple()))
     compile_flags = ["-v"] if verbose else []
-    if jobs:
+    if "jobs" in meson_args:
+        jobs = meson_args["jobs"]
+    if jobs is not None:
         compile_flags += ["-j", str(jobs)]
 
     p = _run(
-        _meson_cli() + ["compile"] + compile_flags + ["-C", build_dir],
+        _meson_cli() + ["compile"] + compile_flags + ["-C", abs_build_dir] + meson_args_compile,
         sys_exit=True,
         output=not quiet,
     )
-    p = _run(
-        _meson_cli()
-        + [
+
+    meson_args_install = list(meson_args.get("install", tuple()))
+    cmd = _meson_cli() + [
             "install",
             "--only-changed",
             "-C",
-            build_dir,
-            "--destdir",
-            install_dir
-            if os.path.isabs(install_dir)
-            else os.path.relpath(abs_install_dir, abs_build_dir),
-        ],
-        output=(not quiet) and verbose,
-    )
+            build_dir
+        ] + meson_args_install
+
+    if "root" not in meson_args:
+        raise KeyError("Project root not specified.")
+
+    dirs_root = meson_args["root"]
+    log_filename = dirs_root / 'meson-install.log'
+    start_time = datetime.datetime.now()
+    cmd_str = ' '.join([str(p) for p in cmd])
+    if meson_args.get("show_build_log", False):
+        ret = subprocess.call(cmd, cwd=dirs_root)
+    else:
+        print("Installing, see meson-install.log...")
+        with open(log_filename, 'w') as log:
+            p = subprocess.Popen(cmd, stdout=log, stderr=log,
+                                    cwd=dirs_root)
+
+        try:
+            # Wait for it to finish, and print something to indicate the
+            # process is alive, but only if the log file has grown (to
+            # allow continuous integration environments kill a hanging
+            # process accurately if it produces no output)
+            last_blip = time.time()
+            last_log_size = os.stat(log_filename).st_size
+            while p.poll() is None:
+                time.sleep(0.5)
+                if time.time() - last_blip > 60:
+                    log_size = os.stat(log_filename).st_size
+                    if log_size > last_log_size:
+                        elapsed = datetime.datetime.now() - start_time
+                        print(f"    ... installation in progress ({elapsed} "
+                                "elapsed)")
+                        last_blip = time.time()
+                        last_log_size = log_size
+
+            ret = p.wait()
+        except:  # noqa: E722
+            p.terminate()
+            raise
```

Comment:
I'd be confused by this behavior on my personal machine I think. I'd rather see this sort of thing in project's CI config I think...